src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -224,8 +224,20 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
       focusSplitPanel: () => splitPanelFocusControl.refs.slider.current?.focus(),
     }));
 
-    const resolvedNavigation = navigationHide ? null : navigation ?? <></>;
     const resolvedStickyNotifications = !!stickyNotifications && !isMobile;
+    //navigation must be null if hidden so toolbar knows to hide the toggle button
+    const resolvedNavigation = useMemo(() => {
```

Comment:
linter recommending I useMemo this hook to prevent exhaustive hooks

Reply:
This makes the linter happy, but does not address the root cause. Why do we even pass `resolvedNavigation` into the `useEffect` below on the line 425?